pkg/cedar/core/client.go

```diff
@@ -75,15 +80,50 @@ type Client struct {
 	auth             runtime.ClientAuthInfoWriter
 	sdk              *apiclient.CEDARCoreAPI
 	hc               *http.Client
+	cache            *cache.Cache
+}
+
+// StartCacheRefresh does the following two things
+// 1. Immediately attempts to populate the cache with current System Summary response data
+// 2. Starts a goroutine that will periodically refresh the cache with new data, based on cacheRefreshTime
+//
+// This function returns no errors, and only logs when something goes wrong
+func (c *Client) StartCacheRefresh(ctx context.Context, cacheRefreshTime time.Duration) {
+	initialErr := c.populateSystemSummaryCache(ctx)
```

Comment:
I am trying to wrap my head around StartCacheRefresh being called as a goroutine in routes.go, while also calling the ticker function as a goroutine inside that goroutine. Would it more or less equivalent to call populateSystemSummaryCache here as a goroutine instead, and call StartCacheRefresh synchronously in routes.go? If I am understanding it right (and I may not be), you would have one less goroutine running continuously.

Reply:
@jstrang-oddball I think the issue with calling `StartCacheRefresh` synchronously is that server startup would block on the initial cache population, which would take a few seconds.

Reply:
These are both super valuable!

This is my first time diving into goroutines, channels, etc., so happy to see this kind of stuff come through as feedback!

I think I like the approach of wrapping the whole thing (initial call and ticker code) in the goroutine, that way we can just synchronously call `StartCacheRefresh`. I'll make some updates and we can loop back!

Reply:
82d22eb42b6d184e343614b4cf640af143caf394